XRDDEV-1075 The API is going to return the latest version of the serv…

Dos4dev opened the pull request:
The API is going to return the latest version of the service title if multiple versions exist.

niis-jenkins commented:
Can one of the admins verify this patch?

iluwatar reviewed (changes requested):
The build is failing due to a couple of Checkstyle problems. Could you fix these? I'll do some further testing tomorrow.

iluwatar commented on src/proxy-ui-api/src/main/java/org/niis/xroad/securityserver/restapi/service/ServiceDescriptionService.java:
```diff
@@ -885,12 +885,16 @@ public String getServiceTitle(ClientType clientType, String serviceCode) {
         return service == null ? null : service.getTitle();
     }
 
+    private Comparator<ServiceType> comparingVersions() {
+        return Comparator.comparing(service -> Integer.valueOf(service.getServiceVersion().replace("v","")));
+    }
+
```
But I understand the problem this pull request tries to solve and I think we are able to solve it with small modifications. Instead of counting on specifically formatted version numbers, we could do a reverse alphabetical comparison. It should produce exactly the same effect.

iluwatar replied:
I would put in some more test data to the `ServiceDescriptionServiceIntegrationTest#getServiceTitle` test and ensure that it behaves as expected with version strings such as `AAA`, `XYZ`, `v1`, V1`. Once that works, I'm sure we can proceed with integrating the change to the X-Road core.

iluwatar commented:
@Dos4dev please find the review comments above. Let me know once you are finished with the changes and I'll take another look.

Dos4dev commented:
>  The build is failing due to a couple of Checkstyle problems. Could you fix these? I'll do some further testing tomorrow.

Hi @iluwatar,
Could you please provide the way how did you receive problems?
I am running `./build_packages.sh` script to be sure that all tests passed.  
Please correct me if I am wrong.

Dos4dev commented:
> @Dos4dev please find the review comments above. Let me know once you are finished with the changes and I'll take another look.

I fixed checkstyle problems and implemented reverse alphabetical comparison for version. I haven't think about alphabetical comparison. Thank you Ilkka :+1:

iluwatar commented:
We appreciate the contribution. Thank you for the collaboration @Dos4dev
